app/Jobs/ImportRockTheVotePosts.php

```diff
@@ -0,0 +1,459 @@
+        }
+
+        return $finalValues;
+    }
+
+    /**
+     * Parse the record for extra details and return them as a JSON object.
+     *
+     * @param  array $record
+     * @param  array $extraData
+     */
+    private function extractDetails($record, $extraData = null)
+    {
+        $details = [];
+
+        $importantKeys = [
+            'Tracking Source',
+            'Started registration',
+            'Finish with State',
+        ];
+
+        foreach ($importantKeys as $key) {
+            $details[$key] = $record[$key];
+        }
+
+        if ($extraData) {
+            $details = array_merge($details, $extraData);
+        }
+
+        return json_encode($details);
+    }
+
+    /**
+     * Translate a status from Rock The Vote into a status that can be sent to Rogue.
+     *
+     * @param  string $rtvStatus
+     * @param  string $rtvFinishWithState
+     * @return string
+     */
+    private function translateStatus($rtvStatus, $rtvFinishWithState)
+    {
+        $rtvStatus = strtolower($rtvStatus);
+        $rtvFinishWithState = strtolower($rtvFinishWithState);
+
+        if($rtvStatus === 'complete') {
+            if ($rtvFinishWithState === "no") {
+                return 'register-form';
+            }
+            if ($rtvFinishWithState === "yes") {
+                return 'register-OVR';
+            }   
+        }
+
+        if (strpos($rtvStatus, 'step') !== false) {
+            return 'uncertain';
+        }
+
+        if ($rtvStatus === 'rejected') {
+            return 'ineligible';
+        }
+
+        return '';
+    }
+
+    /*
+     * Determines if a status should be changed and what it should be changed to.
+     *
+     * @param string $currentStatus
+     * @param string $newStatus
+     * @return string|null
+     */
+    private function updateStatus($currentStatus, $newStatus)
+    {
+        $statusHierarchy = [
+            'uncertain',
+            'ineligible',
+            'confirmed',
+            'register-OVR',
+            'register-form',
+        ];
+
+        $indexOfCurrentStatus = array_search($currentStatus, $statusHierarchy);
+        $indexOfNewStatus = array_search($newStatus, $statusHierarchy);
+
+        return $indexOfCurrentStatus < $indexOfNewStatus ? $newStatus : null;
+    }
+
+    /*
+     * Determines if a record should be process to be stored or if it is not valid.
+     *
+     * @param array $record
+     * @return bool
+    */
+    private function scrubRecord($record)
+    {
+        $isNotValidEmail = strrpos($record['Email address'], 'thing.org') !== false || strrpos($record['Email address'] !== false, '@dosome') || strrpos($record['Email address'], 'rockthevote.com') !== false || strrpos($record['Email address'], 'test') !== false || strrpos($record['Email address'], '+') !== false;
+        $isNotValidLastName = strrpos($record['Last name'], 'Baloney') !== false;
+
+        return $isNotValidEmail || $isNotValidLastName ? false : true;
+    }
+
```

Comment:
and this (can we make a helper)?

Reply:
@chloealee this one differs between Rock the Vote and Turbovote so I think it's gotta stay separate